ethpm/__init__.py

```diff
@@ -3,7 +3,18 @@
 
 ETHPM_DIR = Path(__file__).parent
 ASSETS_DIR = ETHPM_DIR / "assets"
-ETHPM_SPEC_DIR: Path = ETHPM_DIR / "ethpm-spec"
+
+
+def get_ethpm_spec_dir() -> Path:
+    ethpm_spec_dir = ETHPM_DIR / "ethpm-spec"
+    v3_spec = ethpm_spec_dir / "spec" / "v3.spec.json"
+    if not v3_spec.is_file():
+        raise FileNotFoundError(
+            "The ethpm-spec submodule is not available. "
+            "Please import the submodule with `git submodule update --init`"
```

Comment:
I think this error message should be expanded since there are two main paths that it seems like a user would find themselves in this situation.

1. `pip install web3` with a broken package that doesn't include the assets
2. `get clone ...` that doesn't pull the submodules.

Maybe lets expand this error message to include sections for both of those.  In the case where the problem is `get clone` based, the user has agency to fix it.  In the `pip install ...` case, they have no agency to fix the problem because it is likely indicative of a broken release....  

The current exception messaging would be confusing in the `pip install `case since it would not actually fix the user's problem.

Alternatively, we could allow use of an environment variable to point at a different spec directory than the default one... but I'm skeptical that is the right approach because it would only serve as a way to make broken web3.py releases work, when we really should be focusing on not issuing broken releases....

Reply:
@pipermerriam My assumption was that adding the [spec json assets in `MANIFEST.in`](https://github.com/ethereum/web3.py/pull/1682/files#diff-97c91a104c431d0c365565d3ac03ac13R10) (as far as I understand) guarantees that the spec files will always be included in the distribution, so I wasn't really considering case #1 to be a probable scenario. That being said, things can always go wrong I guess, but it just seemed unlikely enough that it doesn't warrant its own message. But, if you feel like this is a case we should guard against, I can definitely update the warning strategy.

Reply:
_queues up `beta.2`_

Reply:
What about someone who does `pip install GIT-HUB_URL`?  I suspect that would put us in that position as well?

Reply:
I just tested a fresh install with `pip install git+https://github.com/ethereum/web3.py/` and the `ethpm-spec` submodule assets are all included in that install. This should mean all the install stories are covered (at least the ones I'm familiar with), but if we want to go with a better safe than sorry approach here, I can update the warning.

Reply:
Interesting.  I would not have expected the submodules assets to make their way in.  In fact, I'm surprised enough by that to be suspicious (but not enough to dig in myself).

If you are confident that there are not any obvious cases where a user would find themselves in this position, I don't think the warning/error-messaging change is necessary.

Reply:
I was also curious, [these docs] seemed the most helpul to me, but I was unable to glean any relevant/meaningful insight from those docs.